src/app/security-group/components/sg-rule-addition-form/sg-rule-addition-form.component.ts

```diff
@@ -0,0 +1,414 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnDestroy, Output, ViewChild } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, FormGroupDirective, NgForm, Validators } from '@angular/forms';
+import { Subscription } from 'rxjs/Subscription';
+import { TranslateService } from '@ngx-translate/core';
+import { ErrorStateMatcher } from '@angular/material';
+
+import { IPVersion, NetworkRuleType } from '../../sg.model';
+import { NetworkProtocol } from '../../network-rule.model';
+import { Utils } from '../../../shared/services/utils/utils.service';
+import {
+  cidrValidator,
+  endPortValidator,
+  icmpV4CodeValidator,
+  icmpV4TypeValidator,
+  icmpV6CodeValidator,
+  icmpV6TypeValidator,
+  startPortValidator
+} from '../../shared/validators';
+import {
+  GetICMPCodeTranslationToken,
+  GetICMPTypeTranslationToken,
+  GetICMPV6CodeTranslationToken,
+  GetICMPV6TypeTranslationToken,
+  IcmpType,
+  icmpV4Types,
+  icmpV6Types
+} from '../../../shared/icmp/icmp-types';
+import 'rxjs/add/operator/startWith';
+import { FirewallRule } from '../../shared/models';
+
+
+function isExist(value: any): boolean {
+  return !!value
+}
+
+function getIpVersionByCidr(cidr: string): IPVersion {
```

Comment:
why can't you use Utils.cidrType directly?

Reply:
I was thinking that will be more clear naming for function. But okay I'll change it